sqlglot/dialects/snowflake.py

```diff
@@ -1277,3 +1285,8 @@ def select_sql(self, expression: exp.Select) -> str:
             if offset and not limit:
                 expression.limit(exp.Null(), copy=False)
             return super().select_sql(expression)
+
+        def put_sql(self, expression: exp.Put) -> str:
+            props = expression.args.get("properties")
+            props_sql = self.properties(props, prefix=" ", sep=" ", wrapped=False) if props else ""
+            return f"PUT '{expression.this}' {expression.args['target']}{props_sql}"
```

Comment:
We need to do `this = self.sql(expression, "this"`; the quotes shouldn't be hard-coded. The same applies to `expression.args["target"]`, because the current approach will silently generate it without a dialect.